### PR TITLE
[JENKINS-31155] Infrastructure for shared libraries

### PR DESCRIPTION
**Pull request by jglick**
Prerequisites of https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/10.

@reviewbybees

### CODE REVIEW

**Comment by abayer** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/cps/GlobalVariable.java [hunk 1]
```diff
@@ -63,17 +66,58 @@
...
+        };
+    }
+
+    /**
+     * Finds a particular variable by name.
+     * @param name see {@link #getName}
+     * @param run see {@link GlobalVariableSet#forRun}
+     * @return the first matching variable, or null if there is none
+     */
+    public static @CheckForNull GlobalVariable byName(@Nonnull String name, @CheckForNull Run<?,?> run) {
```
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/GlobalVariable.java]: This makes me happy. =)

**Comment by abayer** [comment]
re:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScript.java [hunk 2]
```diff
@@ -119,13 +122,12 @@ public final Object invokeMethod(String name, Object args) {
...
     @Override
     public Object getProperty(String property) {
-        for (GlobalVariable v : GlobalVariable.ALL) {
-            if (v.getName().equals(property)) {
-                try {
-                    return v.getValue(this);
-                } catch (Exception x) {
-                    throw new InvokerInvocationException(x);
-                }
+        GlobalVariable v = GlobalVariable.byName(property, $buildNoException());
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/cps/CpsScript.java]: [JENKINS-38167](https://issues.jenkins-ci.org/browse/JENKINS-38167)

**Pull request merged**